File: java/lib/src/main/java/com/svix/BackgroundTaskListOptions.java
package com.xwebhook;

import com.xwebhook.models.BackgroundTaskStatus;
import com.xwebhook.models.BackgroundTaskType;
import com.xwebhook.models.Ordering;

public class BackgroundTaskListOptions extends ListOptions {
    private BackgroundTaskStatus status;
    private BackgroundTaskType task;
    private Ordering order;

    public void setOrder(final Ordering order) {
        this.order = order;
    }

    public Ordering getOrder() {
        return this.order;
    }

    public void setStatus(final BackgroundTaskStatus status) {
        this.status = status;
    }

    public BackgroundTaskStatus getStatus() {
        return this.status;
    }

    public void setTask(final BackgroundTaskType task) {
        this.task = task;
    }

    public BackgroundTaskType getTask() {
        return this.task;
    }
}

File: csharp/Svix/Models/BackgroundTaskListOptions.cs
using System;
using System.Collections.Generic;
using Xwebhook.Model;

namespace Xwebhook.Models
{
    public sealed class BackgroundTaskListOptions : ListOptions
    {
        public BackgroundTaskStatus? Status { get; set; }

        public BackgroundTaskType? Task { get; set; }
    }
}